### PR TITLE
Fix/destructors

### PR DESCRIPTION
**Pull request by SanderBouwman**
Added destructors to all the classes in the logic project.
Removed unused code like fuzzy logic, BSP, etc.
Cleaned up base entity.

### CODE REVIEW

File: src/logic/entity/base_entity.cpp [hunk 1]
```diff
@@ -28,20 +21,14 @@ BaseEntity::BaseEntity(int type, const mesh *base, vec2 position, float mass) {
...
-}
-
-void BaseEntity::update_render_mesh(const mat2 &model) {
-    for (int i = 0; i < _base_representation->size; i++) {
-        _buffer->vectors[i] = _base_representation->vectors[i] * model;
+    if(_representation) {
+        delete _representation;
     }
+    _player = nullptr;
 }
```
Comment (MarkVlamPersie) [inline src/logic/entity/base_entity.cpp]: Waarom check je of _represenetation is gevuld?

Comment (sschrijver) [inline src/logic/entity/base_entity.cpp]: @MarkVlamPersie gaat het goed als je iets wat al nullptr is delete?

Comment (SanderBouwman) [inline src/logic/entity/base_entity.cpp]: Maakt in principe niet zoveel uit.

https://stackoverflow.com/a/9741577

File: src/logic/graph/graph_manager.cpp [hunk 2]
```diff
@@ -15,6 +15,7 @@ GraphManager::~GraphManager() {
     if(graph){
         delete graph;
     }
+    delete _instance;
```
Comment (Rboin) [inline src/logic/graph/graph_manager.cpp]: Enters an infinite loop of trying to delete itself here. Can just set the _instance to nullptr.

File: src/logic/wave/wave_manager.cpp [hunk 3]
```diff
@@ -14,6 +14,11 @@ WaveManager::WaveManager() {
     _wave = nullptr;
 }
 
+WaveManager::~WaveManager() {
+    delete _wave;
+    delete _instance;
```
Comment (Rboin) [inline src/logic/wave/wave_manager.cpp]: Enters infinite loop trying to delete itself.

File: src/logic/world/world.cpp [hunk 4]
```diff
@@ -3,24 +3,24 @@
...
 
 World::World() {}
 
 World::~World() {
-    // Clear world.
+    for(int i = 0; i < _entities.size(); i++){
+        delete _entities.at(i);
+    }
+    delete _representation;
+    delete _instance;
```
Comment (Rboin) [inline src/logic/world/world.cpp]: Same here.

File: src/logic/behaviour/move_order.cpp [hunk 5]
```diff
@@ -31,7 +30,8 @@ MoveOrder *MoveOrder::get_instance() {
 }
 
 MoveOrder::~MoveOrder() {
-    for(int i = 0; i < strategies.size(); i++) {
-        delete strategies.at(i);
+    for(int i = 0; i < _strategies.size(); i++) {
+        delete _strategies.at(i);
     }
+    delete _instance;
```
Comment (Rboin) [inline src/logic/behaviour/move_order.cpp]: Same here

File: src/logic/entity/player_manager.cpp [hunk 6]
```diff
@@ -7,6 +7,14 @@
 
 PlayerManager *PlayerManager::_instance = nullptr;
 
+PlayerManager::~PlayerManager() {
+    for (std::map<int, Player *>::iterator it = _players.begin(); it != _players.end(); it++) {
+        delete (it->second);
+    }
+    _players.clear();
+    delete _instance;
```
Comment (Rboin) [inline src/logic/entity/player_manager.cpp]: Same here

File: src/logic/entity/static/building/building_manager.cpp [hunk 7]
```diff
@@ -12,23 +12,24 @@
...
 BuildingManager::~BuildingManager() {
-    if (buildings.size() > 0) {
-        for (int i = 0; i < buildings.size(); i++) {
-            delete buildings.at(i);
+    if (_buildings.size() > 0) {
+        for (int i = 0; i < _buildings.size(); i++) {
+            delete _buildings.at(i);
         }
     }
+    delete _instance;
```
Comment (Rboin) [inline src/logic/entity/static/building/building_manager.cpp]: Same here

File: src/logic/entity/static/resource_manager.cpp [hunk 8]
```diff
@@ -5,44 +5,46 @@
...
-    if (resources.size() > 0) {
-        for (int i = 0; i < resources.size(); i++) {
-            delete resources.at(i);
+    if (_resources.size() > 0) {
+        for (int i = 0; i < _resources.size(); i++) {
+            delete _resources.at(i);
         }
     }
+    _resources.clear();
+    delete _instance;
```
Comment (Rboin) [inline src/logic/entity/static/resource_manager.cpp]: Also here

**Comment by SanderBouwman** [comment]
Done

File: src/logic/behaviour/orderstrategies/order_strategy.h [hunk 9]
```diff
@@ -2,8 +2,8 @@
 #define CONSTRUCT_AND_DESTROY_ORDER_STRATEGY_BASE_H
 
 #include <vector>
-#include <types.h>
-#include <entity/moving/moving_entity.h>
+#include "types.h"
+#include "entity/moving/moving_entity.h"
```
Comment (sschrijver) [inline src/logic/behaviour/orderstrategies/order_strategy.h]: Shouldn't MovingEntity & BaseEntity be forward declerations?

**Pull request merged**